crates/bevy_tasks/src/single_threaded_task_pool.rs

```diff
@@ -57,27 +58,30 @@ impl TaskPool {
     /// to spawn tasks. This function will await the completion of all tasks before returning.
     ///
     /// This is similar to `rayon::scope` and `crossbeam::scope`
-    pub fn scope<'scope, F, T>(&self, f: F) -> Vec<T>
+    pub fn scope<'env, F, T>(&self, f: F) -> Vec<T>
```

Comment:
This lifetime configuration seems incorrect - if a user manually puts `'static` as the `'env` lifetime, they can get a 'static Scope which is obviously incorrect.

Reply:
made 'env and 'scope invariant on the single threaded task pool. Other lifetime concerns here should be addressed by this comment https://github.com/bevyengine/bevy/pull/4466#issuecomment-1106037526. Please let me know any situations you think might be troublesome and I can try testing them.